File: R/Regression/Simple Linear Regression/simple_linear_regression.R
# Read dataset
dataset = read.csv('Salary_Data.csv')
# dataset = dataset[,2:3]

# Spliting dataset into train and test
library(caTools)
set.seed(123)
split = sample.split(dataset$Salary, SplitRatio = 2/3)
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset, split == FALSE)

# Feature scaling
# training_set[,2:3] = scale(training_set[,2:3])
# test_set[,2:3] = scale(test_set[,2:3])

# Fit Simple Linear Regression
regressor = lm(formula = Salary ~ YearsExperience, data = training_set)
# summary(regressor)

# Predicting test results
y_pred = predict(regressor, newdata = test_set)

# Visualising the training results
library(ggplot2)
ggplot() +
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'red') + 
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
                colour = 'blue') +
  ggtitle('Salary vs Experience (Training Set)') +
  xlab("Years of Experience") +
  ylab('Salary')
# Visualising the test results
ggplot() +
  geom_point(aes(x = test_set$YearsExperience, y = test_set$Salary),
             colour = 'red') + 
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor, newdata = training_set)),
            colour = 'blue') +
  ggtitle('Salary vs Experience (Training Set)') +
  xlab("Years of Experience") +
  ylab('Salary')

File: Python/Regression/Simple Linear Regression/simple_linear_regression.py
# Simple Linear Regression

import numpy as np 
import matplotlib.pyplot as plt 
import pandas as pd

# Read Dataset
dataset = pd.read_csv('Salary_Data.csv')

X = dataset.iloc[:, :-1].values
y = dataset.iloc[:,-1].values


#Spliting the dataset into Train and test 
from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 1/3, random_state = 0)

# Fit the model with training set
from sklearn.linear_model import LinearRegression

regressor = LinearRegression()
regressor.fit(X_train, y_train)

# Predict test data
y_pred = regressor.predict(X_test)

# Visualizing Trained data
plt.scatter(X_train,y_train, c = 'red')
plt.plot(X_train, regressor.predict(X_train), c = 'blue')
plt.title('Salary vs Year of Experience(Training data)')
plt.xlabel('Salary')
plt.ylabel('Years of Experience')
plt.show()


# Visualizing Predicted data
plt.scatter(X_test,y_test, c = 'red')
plt.plot(X_train, regressor.predict(X_train), c = 'blue')
plt.title('Salary vs Year of Experience (Test Data)')
plt.xlabel('Salary')
plt.ylabel('Years of Experience')
plt.show()
